sl_functionality.cc

```diff
@@ -90,6 +90,69 @@ Chirps ServiceLayerFunctionality::monitor(const std::string& username) {
   return chirps;
 }
 
+Chirps ServiceLayerFunctionality::stream(const std::string& username) {
+  std::lock_guard<std::mutex> lock(sl_func_mtx_);
+  // Form stream key to retrieve new chirps broadcast to the user
+  const std::string kStreamKey = "stream::" + username;
+  std::string stream_serial = client_->get(kStreamKey);
+  Chirps chirps;
+  if (stream_serial == kFalse_) {
+    return chirps;
+  }
+  chirps.ParseFromString(stream_serial);
+  // put stream key back into the array as empty
+  client_->put(kStreamKey, kFalse_);
+  return chirps;
+}
+
+void ServiceLayerFunctionality::startStream(const std::string& username, const std::string& hashtag) {
+  std::lock_guard<std::mutex> lock(sl_func_mtx_);
+  if (hashtag.length() == 1 || hashtag.find("#")==std::string::npos) {
+    return;
+  }
+  // Form streaming key for the given hashtag
+  const std::string kStreamingKey = "streaming::" + hashtag;
+  // Retrieve streamers from the key value store
+  std::string streaming_serial = client_->get(kStreamingKey);
+  Streamers streamers;
+  streamers.ParseFromString(streaming_serial);
+
+  // Add the username as a new streamer
+  streamers.add_username(username);
+
+  // put new list in database
+  streamers.SerializeToString(&streaming_serial);
+  client_->put(kStreamingKey, streaming_serial);
+}
+void ServiceLayerFunctionality::endStream(const std::string& username, const std::string& hashtag) {
+  std::lock_guard<std::mutex> lock(sl_func_mtx_);
+  // Form streaming key for the given hashtag
+  const std::string kStreamingKey = "streaming::" + hashtag;
```

Comment:
Move constant string to constant definition? (Here and elsewhere.)

Reply:
I forgot to fork the repository instead of just being on Krishna's repo but done on the forked version (new pr to come I am so sorry)